vars/unitTest.groovy

```diff
@@ -87,9 +90,21 @@ def call(Map config = [:]) {
 
   Map params = [:]
   params['stashes'] = stashes
-  params['script'] = "SSH_KEY_ARGS=${env.SSH_KEY_ARGS} " +
-                     "NODELIST=${nodelist} " +
-                     test_script
+  
+
+  echo "marj> env['WITH_VALGRIND'] = ${env['WITH_VALGRIND']}"
+  lock(label: 'WITH_VALGRIND resource', variable: 'WITH_VALGRIND') {
```

Comment:
This is quite probably not the right solution.  I'm actually not entirely sure what you are trying to do with environment variables here.  Can you try to explain what the goal of using this environment variable here in this groovy step is?